AmplifyPlugins/API/AWSAPICategoryPluginIntegrationTests/GraphQL/GraphQLWithUserPoolIntegrationTests/AuthDirective/GraphQLAuthDirectiveIntegrationTests.swift

```diff
@@ -0,0 +1,244 @@
+            GraphQLAuthDirectiveIntegrationTests.password = password
+
+            let awsConfiguration = try TestConfigHelper.retrieveAWSConfiguration(
+                forResource: GraphQLAuthDirectiveIntegrationTests.awsconfiguration)
+            AWSInfo.configureDefaultAWSInfo(awsConfiguration)
+        } catch {
+            XCTFail("Error during setup: \(error)")
+        }
+    }
+
+    override func setUp() {
+        do {
+            AuthHelper.initializeMobileClient()
+
+            Amplify.reset()
+
+            try Amplify.add(plugin: AWSAPIPlugin())
+
+            let amplifyConfig = try TestConfigHelper.retrieveAmplifyConfiguration(
+                forResource: GraphQLAuthDirectiveIntegrationTests.amplifyConfiguration)
+            try Amplify.configure(amplifyConfig)
+
+            ModelRegistry.register(modelType: SocialNote.self)
+        } catch {
+            XCTFail("Error during setup: \(error)")
+        }
+    }
+
+    override func tearDown() {
+        Amplify.reset()
+        AuthHelper.signOut()
+    }
+
+    /// A model with operations [create, update, delete] allows the model to be read by others but not updated or
+    /// deleted. This creates a read-only model.
+    ///
+    /// - When:
+    ///    - Model restricts access to create, update, delete.
+    /// - Then:
+    ///    - Owner can create and update the model
+    ///    - Others can read the owner's model
+    ///    - Others cannot update or delete the owner's model
+    ///    - Owner can delete the model
+    func testModelIsReadOnly() {
+        AuthHelper.signIn(username: GraphQLAuthDirectiveIntegrationTests.user1,
+                          password: GraphQLAuthDirectiveIntegrationTests.password)
+        let id = UUID().uuidString
+        let content = "owner created content"
+        let ownerCreatedNoteResult = createNote(id, content: content)
+        guard case let .success(ownerCreatedNote) = ownerCreatedNoteResult else {
+            XCTFail("Owner should be able to successfully create a note")
+            return
+        }
+        let ownerReadNoteResult = queryNote(byId: ownerCreatedNote.model.id)
+        guard case let .success(ownerReadNoteOptional) = ownerReadNoteResult,
+            let ownerReadNote = ownerReadNoteOptional else {
+            XCTFail("Owner should be able to query for own note")
+            return
+        }
+        let ownerUpdateNote = SocialNote(id: ownerReadNote.model.id, content: "owner updated content", owner: nil)
+        let ownerUpdatedNoteResult = updateNote(ownerUpdateNote, version: ownerReadNote.syncMetadata.version)
+        guard case let .success(ownerUpdatedNote) = ownerUpdatedNoteResult else {
+            XCTFail("Owner should be able to update own note")
+            return
+        }
+        AuthHelper.signOut()
+        AuthHelper.signIn(username: GraphQLAuthDirectiveIntegrationTests.user2,
+                          password: GraphQLAuthDirectiveIntegrationTests.password)
```

Comment:
Is there any way to validate that we are we are actually logged in as a different user? e.g. validate a change in the userSub or something like that (and fail otherwise)?

Reply:
good idea, this would be good to do when we go to migrate the AuthHelper to use Amplify.Auth, we can improve the method to check 
- if signed in 
- if yes, grab existing userSub then sign out
- and sign in with user2, and check if userSub is updated. 
I would prefer to keep this in mind and do this when we migrate each category to use Amplify.Auth